invenio_rdm_records/requests/user_moderation/actions.py

```diff
@@ -63,8 +43,18 @@ def on_block(user_id, uow=None, **kwargs):
         pass
 
     # soft-delete all the published records of that user
-    for (recid,) in _get_records_for_user(user_id):
-        delete_record.delay(recid, tombstone_data)
+    for recid in get_user_records(user_id):
+        uow.register(TaskOp(delete_record, recid=recid, tombstone_data=tombstone_data))
+
+    # Send cleanup task to make sure all records are deleted
+    uow.register(
+        TaskOp.for_async_apply(
+            user_block_cleanup,
+            kwargs=dict(user_id=user_id, tombstone_data=tombstone_data),
+            # wait for 10 minutes before starting the cleanup
+            countdown=10 * 60,
```

Comment:
This is a bit of a "magic number" here. The idea is that we first send off the individual tasks for each record but also send off a delayed task that will perform a more "thorough" check by querying the DB.